### PR TITLE
Android: Stability fixes and other minor ones

### PR DESCRIPTION
**Pull request by Olekoop**
This PR has a fix that makes Android builds more stable
It also has fixes warning and usage of deprecated functions

### CODE REVIEW

File: Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp [hunk 1]
```diff
@@ -421,6 +421,11 @@ void EventLoopManagerUnix::wait_for_events(EventLoopImplementation::PumpMode mod
     if (error_or_marked_fd_count.value() != 0) {
         // Handle file system notifiers by making them normal events.
         for (size_t i = 1; i < thread_data.poll_fds.size(); ++i) {
+            // FIXME: Make the check work under Android
+#ifdef __ANDROID__
+            auto& notifier = *thread_data.notifier_by_index[i];
+            ThreadEventQueue::current().post_event(notifier, make<NotifierActivationEvent>(notifier.fd(), notifier.type()));
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp:427]: I'm not sure I understand this commit. In the case that we end up using an EventImplementationUnix, we aren't using ALooper at all. We are using our own poll-based event loop here.

Is it possibly the case that we should be using ALooper in all of the helper processes as well?

in any case, if this is the correct hack to make things not freeze up, we should be looking at AK_OS_ANDROID, not `__ANDROID__`

Comment (Olekoop) [reply Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp:427]: Well, it works on my phone. Without the patch it gets into a loop around this section.
Fixed the look up

File: Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
+#/*
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp:1]: Not sure what this # is doing here, seems a typo?

File: Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp (hunk 1 above)
Comment (ADKaster) [reply Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp:427]: Fair enough. Can you update the commit message to reflect that, and create an issue to revisit this?

**Review by ADKaster — COMMENTED** [review]
Thanks for fixing this up! A couple more minor things and then I think this is good to go.

File: Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp (hunk 2 above)
Comment (Olekoop) [reply Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp]: Sorry, fixed it

File: Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp (hunk 1 above)
Comment (Olekoop) [reply Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp:427]: Ok, I have described what I could saw here: #484 and I updated the commit comment

**Review by ADKaster — APPROVED** [review]

**Pull request merged**